pkg/apiserver/statement/statement.go

```diff
@@ -56,7 +56,11 @@ func Register(r *gin.RouterGroup, auth *user.AuthService, s *Service) {
 // @Security JwtAuth
 // @Failure 401 {object} utils.APIError "Unauthorized failure"
 func (s *Service) schemasHandler(c *gin.Context) {
-	db := utils.GetTiDBConnection(c)
+	db, err := utils.GetTiDBConnection(c)
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
```

Comment:
You don't need to change the signature as long as you introduce some authorization like concept to the `MWAuthRequired`, like:

`auth.MWAuthRequired(AuthOptions{RequireTiDB: true})`

When `RequireTiDB` is set to true, KvModeAuth will fail.

Note that there is another UCP for [authorization](https://github.com/pingcap-incubator/tidb-dashboard/issues/89) so a simple authorization is enough here.